tests/rbd_mirror/test_rbd_mirror_shrink_image_primary.py

```diff
@@ -0,0 +1,131 @@
+from tests.rbd.rbd_utils import Rbd
+from tests.rbd_mirror.rbd_mirror_utils import rbd_mirror_config
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def test_image_shrink_primary(rbd_mirror, pool_type, **kw):
+    """
+    Test to verify image size in secondary when shrink in primary
+
+    Args:
+        rbd_mirror: rbd mirror object
+        pool_type: ec pool or rep pool
+        **kw: test data
+        Example::
+            ec_pool_config:
+                mode: image
+                mirrormode: snapshot
+                size: 2G
+                io_total: 10
+
+    Returns:
+        0 - if test case pass
+        1 - it test case fails
+    """
+    try:
+        log.info(kw)
+        mirror1 = rbd_mirror.get("mirror1")
+        mirror2 = rbd_mirror.get("mirror2")
+        config = kw.get("config")
+        pool = config[pool_type]["pool"]
+        image = config[pool_type]["image"]
+        imagespec = pool + "/" + image
+        resize = "1G"
+        mirror1.resize_image(imagespec=imagespec, size=resize)
+
+        if config[pool_type]["mode"] == "image":
+            mirror1.create_mirror_snapshot(imagespec)
+            mirror2.wait_for_status(imagespec=imagespec, state_pattern="up+replaying")
+        else:
+            mirror2.wait_for_status(imagespec=imagespec, state_pattern="up+replaying")
```

Comment:
I think appropriate condition would be to verify that `config[pool_type]["mirrormode"]` is snapshot right ?

I'm not sure why are we waiting for up+replaying here
wouldn't it be the same before shrinking the image ?

Reply:
Just to understand, will the state_pattern be "up+replaying" during time it takes to resize?

Reply:
I think yes, shrinking image wouldn't affect image state right ?

Reply:
> I think appropriate condition would be to verify that `config[pool_type]["mirrormode"]` is snapshot right ?
> 
> I'm not sure why are we waiting for up+replaying here wouldn't it be the same before shrinking the image ?

As per the initial_rbd_config method, we are not setting mirror mode for journal based mirroring, so we can not check as config[pool_type]["mirrormode"]` is snapshot because of the keyerror so went with "mode".

Reply:
Added the new method for snapshot mirroring wait_for_snapshot_complete(), to check for snapshot completion and then check for image size in secondary cluster.
For journal based mirroring using existing wait_for_replay_complete() method to wait for replay completion and then check for image size in secondary cluster